### PR TITLE
L059: Exasol: Allow quotes around passwords in CREATE USER

### PR DESCRIPTION
**Pull request by sti0**
### Brief summary of the change made

L059 removes the quotes around passwords within `CREATE USER` statements in Exasol dialect.

```sql
CREATE USER user_1 IDENTIFIED BY "h12_xhz";

--> after fix
CREATE USER user_1 IDENTIFIED BY h12_xhz;
```

```
L:   1 | P:  34 | L059 | Unnecessary quoted identifier "h12_xhz".
L:   1 | P:  43 | L009 | Files must end with a single trailing newline.
==== fixing violations ====
2 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] ...
Attempting fixes...
Persisting Changes...
```

This fixes this bug and ignores the `password_auth` segment type.

### Are there any other side effects of this change that we should be aware of?
Nope

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/rules/L059.py
Comment (tunetheweb) [inline src/sqlfluff/rules/L059.py:86]: I’m not loving this hard coding to be honest.

At the very least I think we should have this as an `_ignore_types` list so it’s easier to add new types (as we surely will need to in future), rather than really hardcoded within the rule. See here for a similar example:

https://github.com/sqlfluff/sqlfluff/blob/7a204ddccc66cc37c1b23199668600609459e744/src/sqlfluff/rules/L054.py#L87

**Review by tunetheweb — CHANGES_REQUESTED** [review]

File: src/sqlfluff/rules/L059.py
Comment (sti0) [reply src/sqlfluff/rules/L059.py]: Yes, this is a way better approach. Added this.

**Review by tunetheweb — APPROVED** [review]

**Comment by barrywhart** [comment]
@tunetheweb: I see that you approved the PR at some point, but now it shows "not approved", perhaps because of a force push. Can you please take another look and re-approve if appropriate?

**Review by tunetheweb — APPROVED** [review]

**Comment by tunetheweb** [comment]
> @tunetheweb: I see that you approved the PR at some point, but now it shows "not approved", perhaps because of a force push. Can you please take another look and re-approve if appropriate?

Think I approved at same time as @sti0 asked me to re-review so I was down as an “approver” and a “requested approver”. Don’t think it would have blocked merging but approved again now anyway.

**Comment by sti0** [comment]
> > @tunetheweb: I see that you approved the PR at some point, but now it shows "not approved", perhaps because of a force push. Can you please take another look and re-approve if appropriate?
> 
> 
> 
> Think I approved at same time as @sti0 asked me to re-review so I was down as an “approver” and a “requested approver”. Don’t think it would have blocked merging but approved again now anyway.

Yes I clicked the request approve button and at the same time you already approved 😄

**Pull request merged**
